test/performance/environment_test.go

```diff
@@ -14,10 +19,103 @@ func BenchmarkNewHostInfo(b *testing.B) {
 	}
 }
 
+func BenchmarkNewHostInfoWithContext(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		ctx, cncl := context.WithCancel(context.Background())
+		env.NewHostInfoWithContext(ctx, "", nil, "", false)
+		cncl()
+	}
+}
+
 func BenchmarkProceses(b *testing.B) {
 	b.ResetTimer()
 	env := &core.EnvironmentType{}
 	for i := 0; i < b.N; i++ {
 		env.Processes()
 	}
 }
+
+func BenchmarkGetNetworkOverflow(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.GetNetOverflow()
+	}
+}
+
+func BenchmarkGetSystemUUID(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.GetSystemUUID()
+	}
+}
+
+func BenchmarkGetContainerID(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.GetContainerID()
+	}
+}
+
+func BenchmarkIsContainer(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.IsContainer()
+	}
+}
+
+func BenchmarkGetHostname(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.GetHostname()
+	}
+}
+
+func BenchmarkDiskDevices(b *testing.B) {
+	b.ResetTimer()
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.DiskDevices()
+	}
+}
+
+func BenchmarkWriteFiles(b *testing.B) {
+	files := []*proto.File{
+		{
+			Name:        "/tmp/test.conf",
+			Contents:    []byte("contents"),
+			Permissions: "0644",
+		},
+	}
+
+	AllowedDirectoriesMap := map[string]struct{}{"/tmp": {}}
+	backup, err := sdk.NewConfigApplyWithIgnoreDirectives("", nil, []string{})
+	assert.NoError(b, err)
+	b.ResetTimer()
+
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.WriteFiles(backup, files, "/tmp", AllowedDirectoriesMap)
+		os.Remove("/tmp/test.conf")
+	}
+}
+
+func BenchmarkFileStat(b *testing.B) {
+	tempDir := b.TempDir()
+	tempFile := tempDir + "/test.txt"
+	err := os.WriteFile(tempFile, []byte("hello"), 0o644)
+	assert.NoError(b, err)
+	b.ResetTimer()
+
+	env := &core.EnvironmentType{}
+	for i := 0; i < b.N; i++ {
+		env.FileStat(tempFile)
+	}
+	defer os.Remove(tempFile)
```

Comment:
Just a heads up that `b.TempDir()` will remove the dir and all it's files when the test completes. So it is not necessary to remove the files or folders of the test.

Ref: https://pkg.go.dev/testing#B.TempDir( and obviously the same with https://pkg.go.dev/testing#T.TempDir )

```suggestion
```

Reply:
yeah I'll remove it. I explicitly put it in because I was creating it but it seems like unnecessary code alright